src/static/js/modules/jquery/cf_formValidator.js

```diff
@@ -25,39 +25,82 @@ var _validator = {
-  // @param   {string} value  The evaluated value of the input
-  // @returns {object}        The status of each of the tested types
-  _validateInput: function( $input, value ) {
-    var status = {};
+  /**
+   * Checks wether the passed input should be skipped for the passed test type
+   * @param   {object}  elem The input we're testing
+   * @param   {string}  type The test type we're checking for
+   * @returns {boolean}      Wether this input should be tested for the type
+   */
+  _skipType: function( elem, type ) {
+    var typeAttr = elem.attr( 'type' );
+    var skips = {
+      required:   !elem.prop( 'required' ),
+      email:      typeAttr !== 'email',
+      radiogroup: typeAttr !== 'radio',
+      checkgroup: typeAttr !== 'checkbox'
+    };
+    return skips[type];
+  },
 
-    $.each( _validator.settings.types, function() {
-      var validation;
-      var type = this;
-      var notRequired = type === 'required' && !$input.prop( 'required' );
-      var notEmail = type === 'email' && $input.attr( 'type' ) !== 'email';
-      var notRadio = type === 'radiogroup' && $input.attr( 'type' ) !== 'radio';
-      var notCheckbox = type === 'checkgroup' && $input.attr( 'type' ) !== 'checkbox';
+  /**
+   * Validate an individual input for each of the set types
+   * @param   {object} elem The jQuery object of the input
+   * @returns {object}      The status of each of the tested types
+  */
+  _validateTypes: function( elem ) {
+    var status = {};
+    var value = elem.val();
+    var validation = {
+      required: _validate.single( value, { presence: true } ),
+      email:    _validate.single( value, { email: true } )
+    };
 
-      if ( notRequired || notEmail || notRadio || notCheckbox ) {
-        status[type] = null;
+    $.each( _validator.settings.types, function( key, val ) {
+      if ( _validator._skipType( elem, val ) ) {
+        status[val] = null;
         return false;
       }
-
-      if ( type === 'required' ) {
-        validation = _validate.single( value, { presence: true } );
-      } else if ( type === 'email' ) {
-        validation = _validate.single( value, { email: true } );
-      }
-      status[type] = typeof validation === 'undefined';
+      status[val] = typeof validation[val] === 'undefined';
     } );
 
     return status;
   },
 
-  // Validate the fields of our form
-  // @returns {object} Two inner objects containing the valid and invalid fields
+  /**
+   * Returns  the formatted validation of the tested input
+   * @param   {object} elem The input we're testing
+   * @returns {object}      The formatted validation object of the tested input
+   */
+  _validateInput: function( elem ) {
+    return {
+      elem:   elem,
+      value:  elem.val(),
+      label:  $.trim( $( 'label[for="' + elem.attr( 'id' ) + '"]' ).text() ),
+      status: _validator._validateTypes( elem )
+    };
+  },
+
+  /**
+   * Returns  the formatted validation of the tested check group
```

Comment:
`Returns` copy is not necessary here as it can be covered in `@returns` comment on L:86.